src/PostgREST/RangeQuery.hs

```diff
@@ -38,12 +41,13 @@ rangeParse range = do
       rangeIntersection lower upper
     Nothing -> rangeGeq 0
 
-rangeRequested :: RequestHeaders -> NonnegRange
-rangeRequested = rangeParse . fromMaybe "" . lookup hRange
+rangeRequested :: RequestHeaders -> Maybe NonnegRange
+rangeRequested headers = rangeParse <$> lookup hRange headers
 
-restrictRange :: Maybe Integer -> NonnegRange -> NonnegRange
+restrictRange :: Maybe Integer -> Maybe NonnegRange -> Maybe NonnegRange
```

Comment:
If you go back to manipulating straight NonnegRange then this function could return to how it was?